packages/synapse-react-client/src/components/DataGrid/SynapseGrid.tsx

```diff
@@ -43,380 +46,399 @@ import {
   GRID_ROW_REACT_KEY_PROPERTY,
   removeNoOpOperations,
 } from './utils/DataGridUtils'
-import { StartGridSession } from './StartGridSession'
+import { StartGridSession, StartGridSessionHandle } from './StartGridSession'
 import { useDataGridWebSocket } from './useDataGridWebsocket'
 import { useGridUndoRedo } from './hooks/useGridUndoRedo'
 import { applyModelChange, ModelChange } from './utils/applyModelChange'
 import { mapOperationsToModelChanges } from './utils/mapOperationsToModelChanges'
 import { getCellClassName } from './utils/getCellClassName'
 
 export type SynapseGridProps = {
-  query: string
```

Comment:
It was kind of disjointed to pass the query in as a prop, but start the session with an imperative handle call.  Updated this so you just pass the information needed to `initializeGrid`.